src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/world/lighting/MixinSkyLightEngine.java

```diff
@@ -0,0 +1,223 @@
+import io.github.opencubicchunks.cubicchunks.chunk.CubeMapGetter;
+import io.github.opencubicchunks.cubicchunks.chunk.IBigCube;
+import io.github.opencubicchunks.cubicchunks.chunk.LightHeightmapGetter;
+import io.github.opencubicchunks.cubicchunks.chunk.util.CubePos;
+import io.github.opencubicchunks.cubicchunks.mixin.access.common.LayerLightSectionStorageAccess;
+import io.github.opencubicchunks.cubicchunks.utils.Coords;
+import io.github.opencubicchunks.cubicchunks.world.lighting.ICubeLightProvider;
+import io.github.opencubicchunks.cubicchunks.world.lighting.ICubicSkyLightEngine;
+import io.github.opencubicchunks.cubicchunks.world.lighting.ISkyLightColumnChecker;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.core.SectionPos;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.ChunkPos;
+import net.minecraft.world.level.chunk.ChunkStatus;
+import net.minecraft.world.level.levelgen.Heightmap;
+import net.minecraft.world.level.lighting.SkyLightEngine;
+import net.minecraft.world.level.lighting.SkyLightSectionStorage;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(SkyLightEngine.class)
+public abstract class MixinSkyLightEngine extends MixinLayerLightEngine<SkyLightSectionStorage.SkyDataLayerStorageMap, SkyLightSectionStorage> implements ISkyLightColumnChecker,
+    ICubicSkyLightEngine {
+    @Shadow @Final private static Direction[] DIRECTIONS;
+
+    /**
+     * @author CursedFlames
+     * @reason disable vanilla sky light logic
+     */
+    @Inject(method = "checkNode", at = @At("HEAD"), cancellable = true)
+    protected void checkNode(long id, CallbackInfo ci) {
+        if (!this.isCubic) {
+            return;
+        }
+        ci.cancel();
+        super.checkNode(id);
+    }
+
+    /** all parameters are global coordinates */
+    @Override public void checkSkyLightColumn(CubeMapGetter chunk, int x, int z, int oldHeight, int newHeight) {
+        ((LayerLightSectionStorageAccess) this.storage).invokeRunAllUpdates();
+        CubeMap cubeMap = chunk.getCubeMap();
+        int oldHeightCube = Coords.blockToCube(oldHeight - 1);
+        int newHeightCube = Coords.blockToCube(newHeight);
+        if (oldHeight > newHeight) {
+            // not sure if this is necessary - also maybe it should be done inside the loop? not sure if threaded stuff can result in storage becoming out of date inside the loop
+            ((LayerLightSectionStorageAccess) this.storage).invokeRunAllUpdates();
+
+            // TODO cube iteration order might still be important here
+            //      (int y = oldHeight-1; y >= newHeight; y--)
```

Comment:
Is it?

Reply:
Since light updates are queued before being processed, it shouldn't affect it, but I'm not entirely confident about that.

Reply:
We currently don't see any lighting errors. I'd vote for removing that ToDo.